src/Host/Broker/Impl/Sessions/SessionsController.cs

```diff
@@ -54,15 +56,20 @@ public class SessionsController : Controller {
                 interp = _interpManager.Interpreters.First();
             }
 
-            var session = _sessionManager.CreateSession(User.Identity, id, interp, securePassword, profilePath, request.CommandLineArguments);
-            return Task.FromResult<IActionResult>(new ObjectResult(session.Info));
+            try {
+                var session = _sessionManager.CreateSession(User.Identity, id, interp, securePassword, profilePath, request.CommandLineArguments);
+                return Task.FromResult<IActionResult>(new ObjectResult(session.Info));
+            } catch (Exception ex) {
+                return Task.FromResult<IActionResult>(new ApiErrorResult(HttpContext.Response, BrokerApiError.UnableToStartRHost, ex));
+            }
         }
 
         [HttpGet("{id}/pipe")]
         public IActionResult GetPipe(string id) {
             var session = _sessionManager.GetSession(User.Identity, id);
-            if (session?.Process?.HasExited ?? true) {
-                return NotFound();
+            Exception ex = (session?.Process != null && session.Process.HasExited) ? new Win32Exception(session.Process.ExitCode) : null;
```

Comment:
This isn't right - this shouldn't be reporting "unable to start RHost" when connecting to a pipe for a session that doesn't exist, because this endpoint is not responsible for creating sessions. 

So if a session is not in the list, this should still be 404. And there shouldn't be any case where the process has exited, but the session is still in the list, since we remove it in `Process.Exited`.